src/transform/namer.ts

```diff
@@ -92,10 +92,27 @@ export async function namer(session: Session<CodeModel>) {
   const groupParameters = await session.getValue('group-parameters', true);
   model.language.go!.groupParameters = groupParameters;
 
-  // pascal-case and capitzalize acronym names of objects and their fields
+  // fix up type names
+  const structNames = new Set<string>();
+  for (const obj of values(model.schemas.objects)) {
+    obj.language.go!.name = ensureNameCase(obj.language.go!.name);
+    structNames.add(obj.language.go!.name);
+  }
+
+  // fix stuttering type names
+  for (const obj of values(model.schemas.objects)) {
+    const details = <Language>obj.language.go;
+    const originalName = details.name;
+    details.name = trimPackagePrefix(stutteringPrefix, originalName);
+    // if the type was renamed to remove stuttering, check if it collides with an existing type name
+    if (details.name !== originalName && structNames.has(details.name)) {
```

Comment:
I guess this prevents combining the two new loops, because in principal we need to generate all the struct names before we can be certain none collide?

Reply:
Sadly yes.  It's also why some of the code comments had the old type names.